src/main/java/com/uci/inbound/firebase/web/FirebaseWebController.java

```diff
@@ -53,28 +60,128 @@ public class FirebaseWebController {
+                            .fcmToken(message.getReport().getFcmDestAdd())
+                            .userId(message.getReport().getDestAdd())
+                            .build();
+                    saveDeliveryReport(deliveryReport);
+                }
+            } else {
+                log.info("Fetching data from Cassandra... External Id : " + externalId + " : UserId : " + userId);
+                getXMessageDaoCass(externalId, userId).subscribe(new Consumer<List<XMessageDAO>>() {
+                    @Override
+                    public void accept(List<XMessageDAO> xMessageDAOList) {
+                        if (xMessageDAOList != null && xMessageDAOList.isEmpty()) {
+                            log.error("getXMessageDaoCass Dao Empty found : " + xMessageDAOList.size());
+                        }
+                        for (XMessageDAO xMessageDAO : xMessageDAOList) {
+                            DeliveryReport deliveryReport = DeliveryReport.builder()
+                                    .botId(xMessageDAO.getBotUuid().toString())
+                                    .botName(xMessageDAO.getApp())
+                                    .externalId(message.getReport().getExternalId())
+                                    .messageState(message.getEventType())
+                                    .fcmToken(message.getReport().getFcmDestAdd())
+                                    .userId(message.getReport().getDestAdd())
+                                    .build();
+                            saveDeliveryReport(deliveryReport);
+                        }
+                    }
+                });
+            }
+        } else {
+            log.error("FirebaseWebController:deliveryReport:: Invalid Request: " + message);
+        }
+    }
+
+
+    /**
+     * TODO uncomment this function when migrate this with diksha
+     */
+//    @RequestMapping(value = "/web", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+//    public void dikshaWeb(@RequestBody FirebaseWebMessage message) throws JsonProcessingException, JAXBException {
+//
+//        log.info("FirebaseWebController:dikshaWeb:: Request: " + mapper.writeValueAsString(message));
+//
+//        firebaseNotificationAdapter = FirebaseNotificationAdapter.builder()
+//                .botService(botService)
+//                .build();
+//
+//        XMsgProcessingUtil.builder()
+//                .adapter(firebaseNotificationAdapter)
+//                .xMsgRepo(xmsgRepo)
+//                .inboundMessage(message)
+//                .topicFailure(inboundError)
+//                .topicSuccess(inboundProcessed)
+//                .kafkaProducer(kafkaProducer)
+//                .botService(botService)
+//                .redisCacheService(redisCacheService)
+//                .topicOutbound(outboundTopic)
+//                .topicReport(topicReport)
+//                .build()
+//                .process();
+//    }
```

Comment:
What is this in regards to?

Reply:
This is for backup purposes when we revert back to this code.

Reply:
Why do we need to revert back to this?

Reply:
Removed commented code